### PR TITLE
AF-1251: Support to experimental features

### PR DESCRIPTION
**Pull request by pefernan**
Enabling the experimental features framework & adding the experimental CMS data management components

@porcelli @ederign  could you please take a look?

This is a part of an ensemble, please merge with:
https://github.com/kiegroup/appformer/pull/415
https://github.com/kiegroup/kie-wb-common/pull/1965
https://github.com/kiegroup/kie-wb-distributions/pull/779
https://github.com/kiegroup/drools-wb/pull/887
https://github.com/kiegroup/optaplanner-wb/pull/291

### CODE REVIEW

**Comment by pefernan** [comment]
jenkins please execute full downstream build

File: kie-wb-parent/pom.xml [hunk 1]
```diff
@@ -65,7 +65,7 @@
           <configuration>
             <deploy>${project.build.directory}/gwt-symbols-deploy</deploy>
             <localWorkers>${gwt.compiler.localWorkers}</localWorkers>
-            <extraJvmArgs>${gwt.memory.settings} -XX:CompileThreshold=7000 -Derrai.jboss.home=${errai.jboss.home} -Derrai.server.classOutput=${project.build.outputDirectory} -Derrai.dynamic_validation.enabled=true -Dorg.kie.demo=true -Dorg.kie.clean.onstartup=true -Dorg.uberfire.nio.git.ssh.enabled=false -Djava.util.prefs.syncInterval=2000000 -Dorg.uberfire.async.executor.safemode=true -Derrai.marshalling.very_short_names=${errai.marshalling.very_short_names}</extraJvmArgs>
+            <extraJvmArgs>${gwt.memory.settings} -XX:CompileThreshold=7000 -Derrai.jboss.home=${errai.jboss.home} -Derrai.server.classOutput=${project.build.outputDirectory} -Derrai.dynamic_validation.enabled=true -Dorg.kie.demo=true -Dorg.kie.clean.onstartup=true -Dorg.uberfire.nio.git.ssh.enabled=false -Djava.util.prefs.syncInterval=2000000 -Dorg.uberfire.async.executor.safemode=true -Derrai.marshalling.very_short_names=${errai.marshalling.very_short_names} -Dappformer.experimental.features=true</extraJvmArgs>
```
Comment (ederign) [inline kie-wb-parent/pom.xml]: Pere should we enable this on kie-wb parent? This is goes to product right?

**Comment by pefernan** [comment]
jenkins please execute full downstream build

File: kie-wb-parent/pom.xml (hunk 1 above)
Comment (pefernan) [reply kie-wb-parent/pom.xml]: @ederign correct me if wrong, but I think this params only affect when running the app on hosted mode, not the final generated war.

In fact I built the distribution wars and tried them with and without the "appformer.experimental.features" param and worked well.

**Comment by pefernan** [comment]
jenkins please execute full downstream build

**Comment by pefernan** [comment]
jenkins please execute full downstream build

**Comment by pefernan** [comment]
jenkins please execute full downstream build

**Comment by pefernan** [comment]
jenkins execute full downstream build

**Comment by pefernan** [comment]
Jenkins execute full downstream build

**Comment by pefernan** [comment]
unrelated test failures... trying again

**Comment by pefernan** [comment]
Jenkins execute full downstream build

**Comment by pefernan** [comment]
Jenkins execute full downstream build

**Comment by pefernan** [comment]
Jenkins execute full downstream build

**Comment by domhanak** [comment]
Jenkins execute full downstream build

**Comment by domhanak** [comment]
jenkins retest this

**Comment by pefernan** [comment]
Closing

**Pull request closed without merging**